spring-batch-querydsl-reader/src/main/java/org/springframework/batch/item/querydsl/reader/options/QuerydslNoOffsetOptions.java

```diff
@@ -41,7 +41,20 @@ public String getFieldName() {
 
     protected Object getFiledValue(T item) {
         try {
-            Field field = item.getClass().getDeclaredField(fieldName);
+            final Class<?> itemClass = item.getClass();
+            if (itemClass.getSuperclass() != null) {
+                final Class<?> superclass = itemClass.getSuperclass();
+                final Field[] superClassFields = superclass.getDeclaredFields();
+                for (Field field : superClassFields) {
+                    if (field.getName().equals(fieldName)) {
+                        field.setAccessible(true);
+                        superclass.getDeclaredField("id");
+                        return field.get(item);
+                    }
+                }
+            }
+
+            final Field field = itemClass.getDeclaredField("id");
             field.setAccessible(true);
             return field.get(item);
         } catch (NoSuchFieldException | IllegalAccessException e) {
```

Comment:
```suggestion
protected Object getFieldValue(T item) {
        try {
            Class<?> itemClass = item.getClass();
            while (itemClass != null) {
                Field[] fields = itemClass.getDeclaredFields();
                for (Field field : fields) {
                    if (field.getName().equals(fieldName)) {
                        field.setAccessible(true);
                        return field.get(item);
                    }
                }
                itemClass = itemClass.getSuperclass();
            }
            throw new IllegalAccessException(fieldName);
        } catch (IllegalAccessException e) {
            log.error("Not Found or Not Access Field: " + fieldName, e);
            throw new IllegalArgumentException("Not Found or Not Access Field");
        }
    }
```

여러 부모를 상속할 수 있기 때문에 위와 같은 코드를 제안드립니다.